### PR TITLE
Qt/Core: implement max llvm compile threads

### PR DESCRIPTION
**Pull request by Megamouse**

### CODE REVIEW

File: rpcs3/rpcs3qt/settings_dialog.cpp [hunk 1]
```diff
@@ -961,6 +962,13 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> guiSettings, std:
 	xemu_settings->EnhanceCheckBox(ui->spuDebug, emu_settings::SPUDebug);
 	SubscribeTooltip(ui->spuDebug, json_debug["spuDebug"].toString());
 
+	// Comboboxes
+
+	xemu_settings->EnhanceComboBox(ui->maxLLVMThreads, emu_settings::MaxLLVMThreads, true);
+	SubscribeTooltip(ui->maxLLVMThreads, json_debug["maxLLVMThreads"].toString());
+	ui->maxLLVMThreads->setItemText(ui->maxLLVMThreads->findData("0"), tr("All (%1)").arg(QThread::idealThreadCount()));
+	ui->maxLLVMThreads->setMaxCount(QThread::idealThreadCount());
```
Comment (scribam) [inline rpcs3/rpcs3qt/settings_dialog.cpp]: Do you think using `QThread::idealThreadCount()` is better in this context than the function `std::thread::hardware_concurrency()` used in rpcs3/Emu/Cell/PPUThread.cpp? Can't we rely on one unique function to count the maximum number of threads?

**Comment by scribam** [comment]
Tested on Ubuntu: LLVM compilation uses the number of thread configured, works perfeclty :+1:
However, I have two questions related to this PR:
- Is it intentional to put this setting in the Debug tab? I was a bit surprise as I didn't expect to have it there, I thought it would be in the CPU section
- I had a minor issue when I tried to set manually the value of "Max LLVM Compile Threads" in the config.yml file. If I set the value to the maximum number of threads (8 for my computer), save the config file then go to the Debug tab, the value is not "All (8)" but "7". Probably because "All (8)" is 0 in the config file. It is a bit disturbing but I guess people should not mess with the config file like that :slightly_smiling_face:
However, do you think this behaviour could be improved?

**Comment by Megamouse** [comment]
no I don't care for people who think they have to do everything different

**Comment by Megamouse** [comment]
It's too much trouble to move it to another tab right now. I'll think of something when we restructure the dialog again.
But i can change the thread count to go from 1 to x

**Comment by scribam** [comment]
Discussed with others contributors:
- This setting can not be added to the CPU tab considering the lack of space to do it. However, it will be added to the Emulator tab to be accessible to everyone
- The initial behaviour with "0" meaning "All" in the config file will be kept to handle changes such as disabling HT or changing the number of cores enabled which can have an impact on the maximum number of threads

**Pull request merged**